benchmarks/benchmarks/benchmark_algorithms.py

```diff
@@ -31,6 +31,11 @@ def time_betweenness_centrality(self, graph):
         # underlying shortest path methods
         _ = nx.betweenness_centrality(self.graphs_dict[graph])
 
+    def time_eigenvector_centrality_numpy(self, graph):
+        # Added to ensure the connectivity check doesn't affect
+        # performance too much (see gh-6888, gh-7549).
+        _ = nx.eigenvector_centrality_numpy(self.graphs_dict[graph])
+
```

Comment:
Thanks for taking a stab at this @Peiffap + the feedback on benchmarking (it's certainly something that needs to be documented better).

I had been doing this locally as well, and would be in *slight* favor of *not* adding the benchmark in this way for the simple reason that one of the example graphs in this benchmark (i.e. the drug interaction graph) is disconnected, which means running the benchmark with this input fails. Overall this doesn't really have an impact on the benchmarking suite (it ignores the failure and continues running) but I'd be in favor of keeping things "clean" for now.

However - it's not a blocker, I'm perfectly happy for it to go in now and just creating a follow-up issue to keep track of re-doing it!